src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```
@@ -277,34 +433,55 @@ public void hangup(String conferenceId) throws Exception {
-                    return generateCallStatus(getJsonProperty(conferenceCalls.get(0), "id", String.class), CallStatus.CallStatusState.Connected);
+            String deviceDialString = retrieveDeviceDialString();
+            Pattern pattern = Pattern.compile("(\\d+):(\\d+):.+$", Pattern.CASE_INSENSITIVE);
+            Matcher matcher = pattern.matcher(callId);
+            if (!StringUtils.isNullOrEmpty(callId) && matcher.find()) {
+                Integer connectionId = Integer.parseInt(matcher.group(2));
+                try {
+                    JsonNode response = doGet(String.format(CONFERENCE, matcher.group(1)), JsonNode.class);
+                    if (response == null) {
+                        return generateCallStatus(callId, CallStatus.CallStatusState.Disconnected);
+                    }
+                    Boolean conferenceIsActive = getJsonProperty(response, "isActive", Boolean.class);
+                    if (conferenceIsActive != null) {
+                        // Generating callId by given conference response and target connectionId
+                        return generateCallStatus(generateCallId(response, connectionId, deviceDialString),
+                                conferenceIsActive ? CallStatus.CallStatusState.Connected : CallStatus.CallStatusState.Disconnected);
+                    }
+                } catch (CommandFailureException cfe) {
+                    if (cfe.getStatusCode() == 404) {
+                        return generateCallStatus(callId, CallStatus.CallStatusState.Disconnected);
+                    }
+                } catch (UnknownDeviceConnection udc) {
+                    // if connection id provided was not found - logging a warning and moving forward to retrieve the list of active conference calls
+                    logger.warn(String.format("Unable to locate active connection with id %s. Using available connection instead.", connectionId));

```

exception is being swallowed here.

---

I don't know, maybe you have a reason to swallow the exception here...please provide reasoning :)

---

UnknownDeviceConnection is a local exception type, thrown by generateCallId() method, in case if conferenceId was found, but not the connectionId.
Essentially it is one of the valid scenarios, as it would move forward and populate the "Disconnected" status with provided callId